@TestHTTPEndpoint  can not work with RESTEasy reactive

**Describe the bug**
@TestHTTPEndpoint  can not work with RESTEasy reactive .
```
@QuarkusTest
@TestHTTPEndpoint(IExampleResource.class)
public class ExampleResourceTest {

    @Test
    public void testHelloEndpoint() {
        given()
                .when().get("/hello")
                .then()
                .statusCode(200)
                .body(is("Hello RESTEasy Reactive"));
    }

}
```
**Expected behavior**
(Describe the expected behavior clearly and concisely.)

**Actual behavior**
(Describe the actual behavior clearly and concisely.)
```
[ERROR] Errors: 
[ERROR]   ExampleResourceTest.testHelloEndpoint ? Runtime Cannot determine HTTP path for...
[INFO] 
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0

```

**Environment (please complete the following information):**
 - Quarkus version or git rev:   12.0.0.CR1

--- a/docs/src/main/asciidoc/resteasy-reactive.adoc
+++ b/docs/src/main/asciidoc/resteasy-reactive.adoc
@@ -103,6 +103,25 @@ public class Endpoint {
 
 See <<uri-parameters,URI parameters>> for more information about URI mapping.
 
+You can set the root path for all rest endpoints using the `@ApplicationPath` annotation, as shown below.
+
+[source,java]
+----
+package org.acme.rest;
+
+import javax.ws.rs.ApplicationPath;
+import javax.ws.rs.core.Application;
+
+@ApplicationPath("/api")
+public static class MyApplication extends Application {
+
+}
+----
+
+This will cause all rest endpoints to be resolve relative to `/api`, so the endpoint above with `@Path("rest")` would
+be accessible at `/api/rest/. You can also set the `quarkus.rest.path` build time property to set the root path if you
+don't want to use an annotation.
+
 === Declaring endpoints: HTTP methods
 
 Each endpoint method must be annotated with one of the following annotations, which defines which HTTP

--- a/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/ResteasyReactiveProcessor.java
+++ b/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/ResteasyReactiveProcessor.java
@@ -244,7 +244,8 @@ public class ResteasyReactiveProcessor {
             ExceptionMappersBuildItem exceptionMappersBuildItem,
             ParamConverterProvidersBuildItem paramConverterProvidersBuildItem,
             ContextResolversBuildItem contextResolversBuildItem,
-            List<MethodScannerBuildItem> methodScanners) throws NoSuchMethodException {
+            List<MethodScannerBuildItem> methodScanners, ResteasyReactiveServerConfig serverConfig)
+            throws NoSuchMethodException {
 
         if (!resourceScanningResultBuildItem.isPresent()) {
             // no detected @Path, bail out
@@ -475,7 +476,7 @@ public class ResteasyReactiveProcessor {
             BeanFactory<ResteasyReactiveInitialiser> initClassFactory = recorder.factory(QUARKUS_INIT_CLASS,
                     beanContainerBuildItem.getValue());
 
-            String applicationPath = determineApplicationPath(index);
+            String applicationPath = determineApplicationPath(index, serverConfig.path);
             // spec allows the path contain encoded characters
             if ((applicationPath != null) && applicationPath.contains("%")) {
                 applicationPath = Encode.decodePath(applicationPath);
@@ -572,10 +573,10 @@ public class ResteasyReactiveProcessor {
         });
     }
 
-    private String determineApplicationPath(IndexView index) {
+    private String determineApplicationPath(IndexView index, String defaultPath) {
         Collection<AnnotationInstance> applicationPaths = index.getAnnotations(ResteasyReactiveDotNames.APPLICATION_PATH);
         if (applicationPaths.isEmpty()) {
-            return null;
+            return defaultPath;
         }
         // currently we only examine the first class that is annotated with @ApplicationPath so best
         // fail if the user code has multiple such annotations instead of surprising the user

--- a/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/ResteasyReactiveServerConfig.java
+++ b/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/ResteasyReactiveServerConfig.java
@@ -1,0 +1,15 @@
+package io.quarkus.resteasy.reactive.server.deployment;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot(name = "rest")
+public class ResteasyReactiveServerConfig {
+
+    /**
+     * Set this to override the default path for JAX-RS resources if there are no
+     * annotated application classes.
+     */
+    @ConfigItem(defaultValue = "/")
+    String path;
+}

--- a/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/path/HelloResource.java
+++ b/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/path/HelloResource.java
@@ -1,0 +1,13 @@
+package io.quarkus.resteasy.reactive.server.test.path;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+
+@Path("/hello")
+public class HelloResource {
+
+    @GET
+    public String hello() {
+        return "hello";
+    }
+}

--- a/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/path/RestPathTestCase.java
+++ b/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/path/RestPathTestCase.java
@@ -1,0 +1,28 @@
+package io.quarkus.resteasy.reactive.server.test.path;
+
+import org.hamcrest.Matchers;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+import io.restassured.RestAssured;
+
+public class RestPathTestCase {
+
+    @RegisterExtension
+    static QuarkusUnitTest test = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class).addClass(HelloResource.class)
+                    .addAsResource(new StringAsset("quarkus.rest.path=/foo"), "application.properties"));
+
+    @Test
+    public void testRestPath() {
+        RestAssured.get("/hello")
+                .then().statusCode(404);
+        RestAssured.get("/foo/hello")
+                .then().statusCode(200).body(Matchers.equalTo("hello"));
+
+    }
+}

--- a/extensions/resteasy-reactive/quarkus-resteasy-reactive/runtime/src/main/java/io/quarkus/resteasy/reactive/server/runtime/ResteasyReactiveTestHttpProvider.java
+++ b/extensions/resteasy-reactive/quarkus-resteasy-reactive/runtime/src/main/java/io/quarkus/resteasy/reactive/server/runtime/ResteasyReactiveTestHttpProvider.java
@@ -1,0 +1,70 @@
+package io.quarkus.resteasy.reactive.server.runtime;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Optional;
+import java.util.function.Function;
+
+import javax.ws.rs.Path;
+
+import org.eclipse.microprofile.config.ConfigProvider;
+
+import io.quarkus.runtime.test.TestHttpEndpointProvider;
+
+public class ResteasyReactiveTestHttpProvider implements TestHttpEndpointProvider {
+    @Override
+    public Function<Class<?>, String> endpointProvider() {
+        return new Function<Class<?>, String>() {
+            @Override
+            public String apply(Class<?> aClass) {
+                String value = getPath(aClass);
+                if (value == null) {
+                    return null;
+                }
+                if (value.startsWith("/")) {
+                    value = value.substring(1);
+                }
+                //TODO: there is not really any way to handle @ApplicationPath, we could do something for @QuarkusTest apps but we can't for
+                //native apps, so we just have to document the limitation
+                String path = "/";
+                Optional<String> appPath = ConfigProvider.getConfig().getOptionalValue("quarkus.rest.path", String.class);
+                if (appPath.isPresent()) {
+                    path = appPath.get();
+                }
+                if (!path.endsWith("/")) {
+                    path = path + "/";
+                }
+                value = path + value;
+                return value;
+            }
+        };
+    }
+
+    private String getPath(Class<?> aClass) {
+        String value = null;
+        for (Annotation annotation : aClass.getAnnotations()) {
+            if (annotation.annotationType().getName().equals(Path.class.getName())) {
+                try {
+                    value = (String) annotation.annotationType().getMethod("value").invoke(annotation);
+                    break;
+                } catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        }
+        if (value == null) {
+            for (Class<?> i : aClass.getInterfaces()) {
+                value = getPath(i);
+                if (value != null) {
+                    break;
+                }
+            }
+        }
+        if (value == null) {
+            if (aClass.getSuperclass() != Object.class) {
+                value = getPath(aClass.getSuperclass());
+            }
+        }
+        return value;
+    }
+}

--- a/extensions/resteasy-reactive/quarkus-resteasy-reactive/runtime/src/main/resources/META-INF/services/io.quarkus.runtime.test.TestHttpEndpointProvider
+++ b/extensions/resteasy-reactive/quarkus-resteasy-reactive/runtime/src/main/resources/META-INF/services/io.quarkus.runtime.test.TestHttpEndpointProvider
@@ -1,0 +1,1 @@
+io.quarkus.resteasy.reactive.server.runtime.ResteasyReactiveTestHttpProvider

--- a/integration-tests/hibernate-reactive-postgresql/src/test/java/io/quarkus/it/hibernate/reactive/postgresql/HibernateReactiveAlternativeTest.java
+++ b/integration-tests/hibernate-reactive-postgresql/src/test/java/io/quarkus/it/hibernate/reactive/postgresql/HibernateReactiveAlternativeTest.java
@@ -4,6 +4,7 @@ import static org.hamcrest.Matchers.is;
 
 import org.junit.jupiter.api.Test;
 
+import io.quarkus.test.common.http.TestHTTPEndpoint;
 import io.quarkus.test.junit.QuarkusTest;
 import io.restassured.RestAssured;
 
@@ -11,12 +12,13 @@ import io.restassured.RestAssured;
  * Test various JPA operations running in Quarkus when the session is injected without Uni or CompletionStage.
  */
 @QuarkusTest
+@TestHTTPEndpoint(HibernateReactiveTestEndpointAlternative.class)
 public class HibernateReactiveAlternativeTest {
 
     @Test
     public void reactiveFind() {
         RestAssured.when()
-                .get("/alternative-tests/reactiveFind")
+                .get("/reactiveFind")
                 .then()
                 .body(is("{\"id\":5,\"name\":\"Aloi\"}"));
     }
@@ -24,7 +26,7 @@ public class HibernateReactiveAlternativeTest {
     @Test
     public void reactiveFindMutiny() {
         RestAssured.when()
-                .get("/alternative-tests/reactiveFindMutiny")
+                .get("/reactiveFindMutiny")
                 .then()
                 .body(is("{\"id\":5,\"name\":\"Aloi\"}"));
     }
@@ -32,7 +34,7 @@ public class HibernateReactiveAlternativeTest {
     @Test
     public void reactivePersist() {
         RestAssured.when()
-                .get("/alternative-tests/reactivePersist")
+                .get("/reactivePersist")
                 .then()
                 .body(is("Tulip"));
     }
@@ -40,7 +42,7 @@ public class HibernateReactiveAlternativeTest {
     @Test
     public void reactiveRemoveTransientEntity() {
         RestAssured.when()
-                .get("/alternative-tests/reactiveRemoveTransientEntity")
+                .get("/reactiveRemoveTransientEntity")
                 .then()
                 .body(is("OK"));
     }
@@ -48,7 +50,7 @@ public class HibernateReactiveAlternativeTest {
     @Test
     public void reactiveRemoveManagedEntity() {
         RestAssured.when()
-                .get("/alternative-tests/reactiveRemoveManagedEntity")
+                .get("/reactiveRemoveManagedEntity")
                 .then()
                 .body(is("OK"));
     }
@@ -56,7 +58,7 @@ public class HibernateReactiveAlternativeTest {
     @Test
     public void reactiveUpdate() {
         RestAssured.when()
-                .get("/alternative-tests/reactiveUpdate")
+                .get("/reactiveUpdate")
                 .then()
                 .body(is("Tina"));
     }

--- a/integration-tests/hibernate-reactive-postgresql/src/test/java/io/quarkus/it/hibernate/reactive/postgresql/HibernateReactiveTest.java
+++ b/integration-tests/hibernate-reactive-postgresql/src/test/java/io/quarkus/it/hibernate/reactive/postgresql/HibernateReactiveTest.java
@@ -5,6 +5,7 @@ import static org.hamcrest.Matchers.is;
 
 import org.junit.jupiter.api.Test;
 
+import io.quarkus.test.common.http.TestHTTPEndpoint;
 import io.quarkus.test.junit.QuarkusTest;
 import io.restassured.RestAssured;
 
@@ -12,12 +13,13 @@ import io.restassured.RestAssured;
  * Test various JPA operations running in Quarkus
  */
 @QuarkusTest
+@TestHTTPEndpoint(HibernateReactiveTestEndpoint.class)
 public class HibernateReactiveTest {
 
     @Test
     public void reactiveFind() {
         RestAssured.when()
-                .get("/tests/reactiveFind")
+                .get("/reactiveFind")
                 .then()
                 .body(is("{\"id\":5,\"name\":\"Aloi\"}"));
     }
@@ -25,7 +27,7 @@ public class HibernateReactiveTest {
     @Test
     public void reactiveCowPersist() {
         RestAssured.when()
-                .get("/tests/reactiveCowPersist")
+                .get("/reactiveCowPersist")
                 .then()
                 .body(containsString("\"name\":\"Carolina\"}")); //Use containsString as we don't know the Id this object will have
     }
@@ -33,7 +35,7 @@ public class HibernateReactiveTest {
     @Test
     public void reactiveFindMutiny() {
         RestAssured.when()
-                .get("/tests/reactiveFindMutiny")
+                .get("/reactiveFindMutiny")
                 .then()
                 .body(is("{\"id\":5,\"name\":\"Aloi\"}"));
     }
@@ -41,7 +43,7 @@ public class HibernateReactiveTest {
     @Test
     public void reactivePersist() {
         RestAssured.when()
-                .get("/tests/reactivePersist")
+                .get("/reactivePersist")
                 .then()
                 .body(is("Tulip"));
     }
@@ -49,7 +51,7 @@ public class HibernateReactiveTest {
     @Test
     public void reactiveRemoveTransientEntity() {
         RestAssured.when()
-                .get("/tests/reactiveRemoveTransientEntity")
+                .get("/reactiveRemoveTransientEntity")
                 .then()
                 .body(is("OK"));
     }
@@ -57,7 +59,7 @@ public class HibernateReactiveTest {
     @Test
     public void reactiveRemoveManagedEntity() {
         RestAssured.when()
-                .get("/tests/reactiveRemoveManagedEntity")
+                .get("/reactiveRemoveManagedEntity")
                 .then()
                 .body(is("OK"));
     }
@@ -65,7 +67,7 @@ public class HibernateReactiveTest {
     @Test
     public void reactiveUpdate() {
         RestAssured.when()
-                .get("/tests/reactiveUpdate")
+                .get("/reactiveUpdate")
                 .then()
                 .body(is("Tina"));
     }

